sgv2-service-common/src/main/java/io/stargate/sgv2/common/cql/builder/BuiltCondition.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright DataStax, Inc. and/or The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.sgv2.common.cql.builder;
+
+import io.stargate.sgv2.common.cql.ColumnUtils;
+import java.util.Optional;
+import org.immutables.value.Value.Immutable;
+import org.immutables.value.Value.Style;
+import org.immutables.value.Value.Style.ImplementationVisibility;
+
+@Immutable
+@Style(visibility = ImplementationVisibility.PACKAGE)
+public interface BuiltCondition {
```

Comment:
adding Javadocs would be awesome and so helpful for all developers, especially external ones